extension-client/src/command/shell/tfcost/index.ts

```diff
@@ -0,0 +1,22 @@
+import * as shell from '..';
+import * as util from '../../../util';
+var os = require('os');
+
+const TFCOST_COMMAND = 'tfcost';
+
+export function calculateTFCost(key: string): Promise<string | Error> {
+    const apikey = 'IC_API_KEY=';
+    const tfcostCommand = `tfcost plan ${util.workspace.getSecureDirectoryPath()}/plan.json --json`;
+    var command: string;
+    if (os.platform() === 'darwin' || os.platform() === 'linux'){
+        command = `export ${apikey}${key} && ${tfcostCommand}`;
+    }
+    else{
+        command = `set "${apikey}${key}" & call ${tfcostCommand}`;
+    }
+    return shell.execute(command);
+}
+
+export function verifyBinary(): Promise<string | Error> {
```

Comment:
I like your thought process where you created a separate folder for tfcost. 

The code could be further refactored to have meaningful functions. If I were you I would rename `verifyBinary` to `isInstalled`. This way it improves readability, for instance `tfcost.isInstalled()` gives clear message what we are trying to check